draft-irtf-cfrg-frost.md

```diff
@@ -214,22 +215,44 @@ FROST requires the use of a cryptographically secure hash function, generically
 written as H, which functions effectively as a random oracle. For concrete
 recommendations on hash functions which SHOULD be used in practice, see
 {{ciphersuites}}. Using H, we introduce three separate domain-separated hashes,
-H1, H2, and H3, where H1 and H2 map arbitrary byte strings to Scalar elements of
+H1, H2, H3, and H4, where H1, H2, and H4 map arbitrary byte strings to Scalar elements of
 the prime-order group scalar field, and H3 is an alias for H with a domain separator. 
-The details of H1, H2, and H3 vary based on ciphersuite. See {{ciphersuites}}
+The details of H1, H2, H3, and H4 vary based on ciphersuite. See {{ciphersuites}}
 for more details about each.
 
 # Helper functions {#helpers}
 
 Beyond the core dependencies, the protocol in this document depends on the
 following helper operations:
 
+- Nonce generation, {{dep-nonces}}
 - Schnorr signatures, {{dep-schnorr}};
 - Polynomial operations, {{dep-polynomial}};
 - Encoding operations, {{dep-encoding}};
 - Signature binding {{dep-binding-factor}}, group commitment {{dep-group-commit}}, and challenge computation {{dep-sig-challenge}}
 
-This sections describes these operations in more detail.
+These sections describes these operations in more detail.
+
+## Nonce generation {#dep-nonces}
+
+To hedge against a bad RNG, we generate nonces by sourcing fresh randomness and 
+combine with the secret key, to create a domain-separated hash function from 
+the ciphersuite hash function `H`, `H4`:
+
+~~~
+  nonce_generate(secret):
+  
+  Inputs:
+  - secret, a Scalar
+
+  Outputs: nonce, a Scalar
+
+  def nonce_generate(secret):
+    k_enc = G.RandomBytes()
```

Comment:
```suggestion
    k_enc = RandomBytes(32)
```

Anything more than 32 is extreme overkill, so I'd recommend simplifying here.

Reply:
Don't we have to mention / define a method like this in the context of the document?

Reply:
Should I put this in `terminology`?

Reply:
Yeah, we can put it in the beginning of the helper section. I just don't think it makes much sense as a member function for the group.

Reply:
Yup definitely

Reply:
OH. We were going to vary this by ciphersuite to match the byte size of the scalars. Is 32 bytes fine 'for everyone', in the failure case?